### PR TITLE
BUG: Fetch to local when print object data

### LINKED ISSUE #597: BUG:Encountered some issues with read_feather.
### Describe the bug
Encountered some issues with read_feather

### To Reproduce

1. Python version: python 3.8.9
2. The version of Xorbits: xorbits 0.4.2
3. numpy version :1.24.4 , pandas version :1.5.2
4. Full stack of the error:

D:\python\lib\site-packages\xorbits\core\utils\fallback.py:35: RuntimeWarning: xorbits.pandas.read_feather will fallback to Pandas
  warnings.warn(warning_str, RuntimeWarning)
D:\python\lib\site-packages\xorbits\_mars\deploy\oscar\session.py:1873: UserWarning: No existing session found, creating a new local session now.
  warnings.warn(warning_msg)
2023-07-16 20:40:52,229 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
2023-07-16 20:40:52,340 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
2023-07-16 20:40:52,444 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
2023-07-16 20:40:52,498 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
2023-07-16 20:40:52,525 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
2023-07-16 20:40:52,749 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
2023-07-16 20:40:53,372 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
2023-07-16 20:40:53,538 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
2023-07-16 20:40:53,924 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
2023-07-16 20:40:53,925 xorbits._mars.deploy.oscar.local 20588 WARNING  Web service started at http://127.0.0.1:54753
  0%|          |   0.00/100 [00:00<?, ?it/s]2023-07-16 20:40:54,401 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
2023-07-16 20:40:54,589 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
  0%|          |   0.00/100 [00:01<?, ?it/s]2023-07-16 20:40:55,425 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
2023-07-16 20:40:55,630 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
100%|██████████| 100.00/100 [00:02<00:00, 49.85it/s]
Object <op=RemoteFunction, key=fc57f9387598e612db2cbc1c5be83979_0>
2023-07-16 20:40:56,466 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
2023-07-16 20:40:56,673 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
2023-07-16 20:40:57,498 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
2023-07-16 20:40:57,705 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
2023-07-16 20:40:58,513 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
2023-07-16 20:40:58,751 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
2023-07-16 20:40:59,544 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
2023-07-16 20:40:59,782 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
2023-07-16 20:41:00,573 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
2023-07-16 20:41:00,812 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
2023-07-16 20:41:01,616 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported
2023-07-16 20:41:01,839 xorbits._mars.resource 20588 WARNING  Unexpected error happened when gathering stats of device 0
Traceback (most recent call last):
  File "D:\python\lib\site-packages\xorbits\_mars\resource.py", line 398, in cuda_card_stats
    device_status = nvutils.get_device_status(device_idx)
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 458, in get_device_status
    _nvml_check_error(
  File "D:\python\lib\site-packages\xorbits\_mars\lib\nvutils.py", line 234, in _nvml_check_error
    raise NVMLAPIError(_error_str.decode(), errno=result)
xorbits._mars.lib.nvutils.NVMLAPIError: (3) Not Supported

6. Minimized code to reproduce the error.
import xorbits.numpy as np
import xorbits.pandas as pd

data_path = "./data/"
close_1m = pd.read_feather(data_path + "close_1m.feather")

### PR DESCRIPTION
**Pull request by aresnow1**
## What do these changes do?

- Fetch to local when print object data
- Fix unexpected logs and suppress warnings

## Related issue number

Fixes #597 

## Check code requirements

- [x] tests added / passed (if needed)
- [x] Ensure all linting tests pass

### CODE REVIEW

File: python/xorbits/_mars/utils.py [hunk 1]
```diff
@@ -1676,7 +1676,12 @@ def _get_func_token_values(func):
             tokens.extend([func.args, func.keywords])
             func = func.func
         if hasattr(func, "__code__"):
-            tokens.extend(_get_func_token_values(func))
+            if len(func.__code__.co_code) > 0:
+                tokens.extend(_get_func_token_values(func))
+            else:
+                # for new python version, cython function has __code__,
+                # but it's an empty string.
+                tokens.extend([func.__module__, func.__name__])
```
Comment (codingl2k1) [inline python/xorbits/_mars/utils.py:1684]: I think `func.__qualname__` is better than `func.__name__` for generating token.
e.g.
```python
def foo():
    def bar():
        return
    return bar

b = foo()
b.__name__
'bar'
b.__qualname__
'foo.<locals>.bar'
```

File: python/xorbits/_mars/utils.py [hunk 2]
```diff
@@ -1664,7 +1664,7 @@ def get_func_token(func):
 
 
 def _get_func_token_values(func):
-    if hasattr(func, "__code__"):
+    if hasattr(func, "__code__") and len(func.__code__.co_code) > 0:
```
Comment (codingl2k1) [inline python/xorbits/_mars/utils.py:1667]: Could `co_code` be None?

File: python/xorbits/_mars/services/cluster/uploader.py [hunk 3]
```diff
@@ -162,7 +162,7 @@ async def upload_node_info(self, status: NodeStatus = None):
                 except ValueError:
                     pass
         except RuntimeError as ex:  # pragma: no cover
-            if "cannot schedule new futures" not in str(ex):
+            if "cannot schedule new futures" in str(ex):
```
Comment (codingl2k1) [inline python/xorbits/_mars/services/cluster/uploader.py:165]: Nice catch!

File: python/xorbits/_mars/utils.py (hunk 1 above)
Comment (aresnow1) [reply python/xorbits/_mars/utils.py:1684]: To keep the previous behavior, use module and name for generating token.

File: python/xorbits/_mars/utils.py (hunk 2 above)
Comment (aresnow1) [reply python/xorbits/_mars/utils.py:1667]: In latest python version, cython function's co_code is an empty string.

**Review by codingl2k1 — APPROVED** [review]
LGTM

**Review by ChengjieLi28 — APPROVED** [review]

**Pull request merged**